pkg/oceancd/model/operator/install.go

```diff
@@ -0,0 +1,163 @@
+					Annotated: configs.Annotated{Annotations: map[string]string{}},
+				},
+				ExtraEnv:                  []corev1.EnvVar{},
+				NodeSelector:              map[string]string{},
+				Tolerations:               []corev1.Toleration{},
+				ImagePullSecrets:          []corev1.LocalObjectReference{},
+				ServiceAccountAnnotations: map[string]string{},
+			},
+		},
+		ArgoRolloutsConfig: configs.ArgoRolloutsConfig{
+			General: configs.ArgoRolloutsGeneralConfig{
+				Namespaced:         configs.Namespaced{Namespace: "argo-rollouts"},
+				Labeled:            configs.Labeled{Labels: map[string]string{}},
+				PodAnnotations:     map[string]string{},
+				PodLabels:          map[string]string{},
+				ServiceAnnotations: map[string]string{},
+				ServiceLabels:      map[string]string{},
+			},
+			Controller: configs.ArgoRolloutsControllerConfig{
+				Replicas:                 pointer.Int64(1),
+				NodeSelector:             map[string]string{},
+				Tolerations:              []corev1.Toleration{},
+				ExtraArgs:                map[string]string{},
+				ExtraEnv:                 []corev1.EnvVar{},
+				ImagePullSecrets:         []corev1.LocalObjectReference{},
+				ContainerSecurityContext: &corev1.SecurityContext{},
+				Probes: configs.Probes{
+					LivenessProbe: &corev1.Probe{
+						ProbeHandler: corev1.ProbeHandler{
+							HTTPGet: &corev1.HTTPGetAction{
+								Path: "/healthz",
+								Port: intstr.FromString("health"),
+							},
+						},
+						InitialDelaySeconds: 30,
+						PeriodSeconds:       20,
+						FailureThreshold:    3,
+						SuccessThreshold:    1,
+						TimeoutSeconds:      10,
+					},
+					ReadinessProbe: &corev1.Probe{
+						ProbeHandler: corev1.ProbeHandler{
+							HTTPGet: &corev1.HTTPGetAction{
+								Path: "/metrics",
+								Port: intstr.FromString("metrics"),
+							},
+						},
+						InitialDelaySeconds: 15,
+						PeriodSeconds:       5,
+						FailureThreshold:    3,
+						SuccessThreshold:    1,
+						TimeoutSeconds:      4,
+					},
+				},
+			},
+			Dashboard: configs.ArgoRolloutsDashboardConfig{
+				Enabled:                  false,
+				Replicas:                 pointer.Int64(1),
+				NodeSelector:             map[string]string{},
+				Tolerations:              []corev1.Toleration{},
+				ExtraEnv:                 []corev1.EnvVar{},
+				ImagePullSecrets:         []corev1.LocalObjectReference{},
+				ContainerSecurityContext: &corev1.SecurityContext{},
+			},
+		},
+	}
+}
+
+type InstallationPayload struct {
+	Namespace string        `json:"namespace"`
+	Manager   ManagerConfig `json:"manager"`
+}
+
+func NewInstallationPayload(config *InstallationConfig) InstallationPayload {
+	return InstallationPayload{Namespace: config.OceanCDConfig.Namespace, Manager: config.OceanCDConfig.ManagerConfig}
+}
+
+type InstallationOutput struct {
+	Manifests []string `json:"manifests"`
+}
+
+type ManifestSet struct {
+	Appliable []string `json:"apply"`
```

Comment:
why do you use another word than `apply`? same for patch?

Reply:
because `apply`is a verb but the fields holds entities that should be applied

Reply:
As I see it, these are manifests we need to apply
OceanCD tells you what you need to apply and what you need to patch
it doesn't tell you that these are appliable and you can decide whether you apply them or not

That's why we chose to use apply in the response from OceanCD and I think for consistency lets use the same names